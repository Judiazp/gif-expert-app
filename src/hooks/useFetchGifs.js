import { useEffect, useState } from "react"
import { getGifs } from '../helpers/getGifs';


export const useFetchGif = ( category ) => {

    const [state, setState] = useState({
        data: [],
        loading: true
    });

    useEffect( () => {
        getGifs( category ).then( gifs => {
            setState({
                data: gifs,
                loading: false
            });
        } )
    }, [ category ])

    return state;
}