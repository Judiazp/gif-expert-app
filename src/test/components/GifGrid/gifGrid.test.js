import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifGrid from '../../../components/gifGrid/'
import { useFetchGif } from '../../../hooks/useFetchGifs';
jest.mock('../../../hooks/useFetchGifs')

describe('Pruebas en <GifGrid />', () => {

    const category = 'Una categoria'
    
    test('Debe mostrar el componente <GifGrid />', () => {
        
        useFetchGif.mockReturnValue({
            data: [],
            loading: true
        })
        const wrapper = shallow( <GifGrid category={ category } />)
        expect( wrapper ).toMatchSnapshot()
    }); 

    test('debe mostrar items cuando se cargan imagenes con useFetch', () => {

        const imgs = [{
            id: 'ABC',
            url: 'https://directionImages/cualquiera/algo.jpg',
            title: 'Un title'
        }]

        useFetchGif.mockReturnValue({
            data: imgs,
            loading: false
        })

        const wrapper = shallow( <GifGrid category={ category } />)

        expect( wrapper ).toMatchSnapshot()
        expect( wrapper.find('p').exists() ).toBe(false)
        expect( wrapper.find('GifCard').length ).toBe( imgs.length )
    });
})