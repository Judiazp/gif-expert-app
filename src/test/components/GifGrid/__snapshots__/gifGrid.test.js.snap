// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Debe mostrar el componente <GifGrid /> 1`] = `
<Fragment>
  <p>
    Loading...
  </p>
  <div
    align="center"
  >
    <h2
      className="title-gifGrid animate__animated animate__fadeInDown"
    >
      Una categoria
    </h2>
  </div>
  <div
    className="grid animate__animated  animate__fadeInUp"
  />
</Fragment>
`;

exports[`Pruebas en <GifGrid /> debe mostrar items cuando se cargan imagenes con useFetch 1`] = `
<Fragment>
  <div
    align="center"
  >
    <h2
      className="title-gifGrid animate__animated animate__fadeInDown"
    >
      Una categoria
    </h2>
  </div>
  <div
    className="grid animate__animated  animate__fadeInUp"
  >
    <GifCard
      key="ABC"
      title="Un title"
      url="https://directionImages/cualquiera/algo.jpg"
    />
  </div>
</Fragment>
`;
